Cubical/HITs/Sn/Properties.agda

```diff
@@ -13,29 +12,329 @@ open import Cubical.Foundations.Prelude
 open import Cubical.Foundations.GroupoidLaws
 open import Cubical.Foundations.Univalence
 open import Cubical.HITs.S1
-open import Cubical.HITs.S3
-open import Cubical.Data.Nat
-open import Cubical.Data.Prod
+open import Cubical.Data.Nat hiding (elim)
+open import Cubical.Data.Sigma
 open import Cubical.HITs.Sn.Base
 open import Cubical.HITs.Susp
-open import Cubical.Data.Unit
-open import Cubical.HITs.Join
-open import Cubical.HITs.Pushout
-open import Cubical.HITs.SmashProduct
-open import Cubical.HITs.PropositionalTruncation
+open import Cubical.HITs.Truncation
+open import Cubical.Homotopy.Loopspace
+open import Cubical.Homotopy.Connected
 
-open Iso
+sphereConnected : (n : HLevel) → isConnected (suc n) (S₊ n)
+sphereConnected n = ∣ ptSn n ∣ , (elim (λ _ → isOfHLevelPath (suc n) (isOfHLevelTrunc (suc n)) _ _)
+                                      (λ a → sym (spoke ∣_∣ (ptSn n)) ∙ spoke ∣_∣ a))
+
+-- Elimination principles for spheres
+sphereElim : ∀ {ℓ} (n : ℕ) {A : (S₊ (suc n)) → Type ℓ} → ((x : S₊ (suc n)) → isOfHLevel (suc n) (A x))
+          → A (ptSn (suc n))
+          → (x : S₊ (suc n)) → A x
+sphereElim zero hlev pt = toPropElim hlev pt
+sphereElim (suc n) hlev pt north = pt
+sphereElim (suc n) {A = A} hlev pt south = subst A (merid (ptSn (suc n))) pt
+sphereElim (suc n) {A = A} hlev pt (merid a i) = helper i
+  where
+  helper : PathP (λ i → A (merid a i)) pt (subst A (merid (ptSn (suc n))) pt)
```

Comment:
Is helper really that helpful here? It seems very clear what the type of this goal should be by the two cases above